File: app/src/main/java/com/namazed/beatboxbignerdranch/Sound.java
package com.namazed.beatboxbignerdranch;

public class Sound {
    private String assetPath;
    private String name;
    private int soundId;

    Sound(String assetPath) {
        this.assetPath = assetPath;
        String[] components = assetPath.split("/");
        String filename = components[components.length - 1];
        name = filename.replace(".wav", "");
    }

    String getAssetPath() {
        return assetPath;
    }

    public String getName() {
        return name;
    }

    int getSoundId() {
        return soundId;
    }

    void setSoundId(int soundId) {
        this.soundId = soundId;
    }
}

File: app/src/main/java/com/namazed/beatboxbignerdranch/Sound.kt
@file: JvmName("SoundKt")
package com.namazed.beatboxbignerdranch

class Sound constructor(internal val assetPath: String) {

    @JvmField
    val name: String

    @JvmField
    var soundId: Int = 0

    init {
        val components = assetPath.split("/".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        val filename = components[components.size - 1]
        name = filename.replace(".wav", "")
    }
}